edkrepo/common/common_repo_functions.py

```diff
@@ -974,12 +974,23 @@ def apply_patchset_operations(repo, operations_list, global_manifest_path, remot
                             try:
                                 repo.git.execute(['git', 'fetch', operation.source_remote, operation.source_branch])
                             except:
-                                raise EdkrepoFetchBranchNotFoundException(FETCH_BRANCH_DOES_NOT_EXIST.format(operation.source_branch))
+                                try:
+                                    repo.git.execute(['git', 'fetch', operation.source_remote, operation.source_branch, '--refetch'])
+                                except:
+                                    try:
+                                        repo.git.execute(['git', 'remote', 'remove', operation.source_remote])

```

Comment:
Instead of adding an except here to remove the remote. Let's use a finally: across the whole block.